math/mathcore/inc/Fit/FitUtil.h

```diff
@@ -478,27 +475,27 @@ namespace FitUtil {
          auto mapFunction = [ &, p](const unsigned i) {
             T W{};
             T W2{};
-            T x1{};
-
-            vecCore::Load<T>(x1, data.GetCoordComponent(i*vecSize,0));
+            T fval{};
 
-            const T * x = nullptr;
             if(data.NDim() > 1) {
-               std::vector<T> xc;
-               xc.resize(data.NDim());
-               xc[0] = x1;
-               for (unsigned int j = 1; j < data.NDim(); ++j)
-                  vecCore::Load<T>(xc[j], data.GetCoordComponent(i * vecSize, j));
-               x = xc.data();
+               std::vector<T> x(data.NDim());
+               for (unsigned int j = 0; j < data.NDim(); ++j)
+                  x[j] = *data.GetCoordComponent(i, j);
+#ifdef USE_PARAMCACHE
+                fval = func ( x.data() );
+#else
+                fval = func ( x.data(), p );
+#endif
```

Comment:
I don't really like the `#ifdef` here and elsewhere. This is a nighmare for testing and maintaining. Isn't it possible to just define the function to always take two parameters, but ignore the second one when `USE_PARAMCACHE` is or is not defined? The logic looks backward here (looks like a parameter cache is passed when `USE_PARAMCACHE` is *not* defined).